### ISSUE TITLE
Level rather than level-offset

### ISSUE THREAD

**Issue by Buzzardo** [asciidoctor/asciidoctor#3241]
We'd like to be able to specify the level of an included document rather than specify the level-offset. We have cases where we use the same content in more than one parent document and at different levels. Rather than calculate the level-offset for the included document, we'd like to specify that the included content starts at (for example) level 2 (==) in one document and level 3 (===) in another document. That would save the trouble of figuring out the offset value and be simpler for us.

In other words, we want to have "offset" as a feature that is parallel to "level-offset".

Thanks.

**Comment by mojavelinux** [comment]
Given how the built-in include processor works, this would be very difficult to support. That's because the built-in include processor doesn't know about the structure of the document it's including (nor is it aware of the structure of the document into which it's being included).

You could support something like this using a custom include processor, though. In a custom processor, you're free to examine the target file, make a decision about how to translate between the level attribute and the leveloffset attribute, then include the content accordingly.

(And #3296 will make your life as an extension writer a whole lot simpler).

**Comment by Alberth289346** [comment]
My comment may be a bit too elaborate on how the current `:leveloffset:` fails to scale to large documents, but it concludes just like the OP that the solution seems to be to use the main document as the one and only place where you specify the desired header level of an included document.

Next to that, I describe how to compute a relative `:leveloffset:` for the included file from the desired level and the first header in the included file, such that the desired level is realized in the final document for the entire included file including nested included files.

I write that in a more algorithmic way, and in way that seems closer to the current asciidoctor implementation, for as far as I can see, given my lack of knowledge in the Ruby language.

----
The python script shows how to create work-around solution that has the same effect. I agree this is not what you want to have, but in case of not understanding the suggested way of implementing the feature, it serves as a common ground, since we fully agree on how `:leveloffset:` behaves.

**Comment by mojavelinux** [comment]
> to use the main document as the one and only place where you specify the desired header level of an included document.

That's just not the case. The `leveloffset` attribute works anywhere in the document.

>  I describe how to compute a relative :leveloffset: for the included file from the desired level and the first header in the included file, such that the desired level is realized in the final document for the entire included file including nested included files.

Were just not going to do that. The leveloffset has no correlation to include files (other than it can be set on the include directive as a shorthand).

If we do support a forced level, then it will be processed at the time the header is processed, just like leveloffset. The meaning would just be slightly different.

> The python script shows how to create work-around solution that has the same effect.

And that should be shared in the project chat, not here. It's a workaround and therefore has no place in the comments for changing the behavior.

**Comment by mojavelinux** [comment]
The other issue here is that this change has to be first defined in the AsciiDoc Language project, not here. We cannot change the behavior of AsciiDoc parsing without involving the AsciiDoc Language project. I have labeled the issue accordingly.
